src/antelope/stores/rex.ts

```diff
@@ -5,28 +5,161 @@
+import EVMChainSettings from 'src/antelope/chains/EVMChainSettings';
+
+
+
+export interface EvmRexDeposit {
+    amount: ethers.BigNumber;
+    until: ethers.BigNumber;
+}
+
+export interface RexModel {
+    withdrawable: ethers.BigNumber;
+    balance: ethers.BigNumber;
+    totalStaking: ethers.BigNumber;
+}
+
+export interface NativeRexModel extends RexModel {
+    _: boolean; // dummy property to avoid lint issues
+}
+
+export interface EvmRexModel extends RexModel {
+    withdrawable: ethers.BigNumber;
+    deposits: EvmRexDeposit[];
+    balance: ethers.BigNumber;
+    totalStaking: ethers.BigNumber;
+}
+
 
 export interface RexState {
-    __: string;
+    __rexData: { [label: Label]: RexModel };
 }
 
 const store_name = 'rex';
 
 export const useRexStore = defineStore(store_name, {
     state: (): RexState => (rexInitialState),
     getters: {
+        getRexData: state => (label: string) => state.__rexData[label] ?? null,
     },
     actions: {
         trace: createTraceFunction(store_name),
-        init: createInitFunction(store_name),
+        init: () => {
+            useFeedbackStore().setDebug(store_name, isTracingAll());
+            getAntelope().events.onAccountChanged.pipe(
+                filter(({ label, account }) => !!label && !!account),
+            ).subscribe({
+                next: async ({ label, account }) => {
+                    if (label === 'current') {
+                        await useRexStore().updateRexDataForAccount(label, toRaw(account));
+                    }
+                },
+            });
+        },
+        async getContractInstance(label: string, address: string) {
+            const authenticator = useAccountStore().getEVMAuthenticator(label);
+            if (!authenticator) {
+                this.trace('getStakedSystemContractInstance', label, '-> no authenticator');
+                throw new AntelopeError('antelope.chain.error_no_default_authenticator');
+            }
+            const contract = await useEVMStore().getContract(authenticator, address);
+            if (!contract) {
+                this.trace('getStakedSystemContractInstance', label, '-> no contract');
+                throw new AntelopeError('antelope.chain.error_no_default_authenticator');
+            }
+            const contractInstance = await contract.getContractInstance();
+            return contractInstance;
+        },
+        async getStakedSystemContractInstance(label: string) {
+            const address = (useChainStore().getChain(label).settings as EVMChainSettings).getStakedSystemToken().address;
+            return this.getContractInstance(label, address);
+        },
+        async getEscrowContractInstance(label: string) {
+            const address = (useChainStore().getChain(label).settings as EVMChainSettings).getEscrowContractAddress();
+            return this.getContractInstance(label, address);
+        },
+        async updateTotalStaking(label: string) {
```

Comment:
can you leave a comment explaining what total staking actually is? I am unclear on what exactly it means